DQM/BeamMonitor/plugins/BeamSpotDipServer.cc

```diff
@@ -0,0 +1,740 @@
+#include "DQM/BeamMonitor/plugins/BeamSpotDipServer.h"
+
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/Framework/interface/LuminosityBlock.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/ConsumesCollector.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+
+//#include "FWCore/Utilities/interface/ESInputTag.h"
+//#include "FWCore/Utilities/interface/Transition.h"
+
+#include "DataFormats/Scalers/interface/DcsStatus.h"
+
+#include <iostream>
+#include <fstream>
+#include <vector>
+#include <ctime>
+#include <sys/stat.h>
+
+#include "Dip.h"
+#include "DipFactory.h"
+#include "DipPublication.h"
+#include "DipTimestamp.h"
+
+using namespace std;
+
+/*****************************************************************************/ 
+class ErrHandler : public DipPublicationErrorHandler {
+ public:
+  virtual ~ErrHandler() = default;
+ private:
+  void handleException(DipPublication* publication, DipException& e) {
+    cerr << "[Dip] exception (at create): " << e.what() << endl;
+  }
+};
+
+/*****************************************************************************/ 
+BeamSpotDipServer::BeamSpotDipServer(const edm::ParameterSet& ps)
+{
+  //
+  verbose = ps.getUntrackedParameter<bool>("verbose");
+  testing = ps.getUntrackedParameter<bool>("testing");
+
+  subjectCMS = ps.getUntrackedParameter<string>("subjectCMS"); 
+  subjectLHC = ps.getUntrackedParameter<string>("subjectLHC"); 
+  subjectPV  = ps.getUntrackedParameter<string>("subjectPV" ); 
+
+  readFromNFS = ps.getUntrackedParameter<bool>("readFromNFS");
+
+  sourceFile  = ps.getUntrackedParameter<string>("sourceFile" ); // beamspot
+  sourceFile1 = ps.getUntrackedParameter<string>("sourceFile1"); // tk status
+
+  timeoutLS = ps.getUntrackedParameter<vector<int>>("timeoutLS");
+
+  //
+  bsLegacyToken_ = esConsumes<edm::Transition::EndLuminosityBlock>();
+
+  dcsStatus_ = consumes<DcsStatusCollection>(
+    ps.getUntrackedParameter<string>("DCSStatus", "scalersRawToDigi")); 
+
+  //
+  dip = Dip::create("CmsBeamSpotServer");
+  cerr << "[Dip] reading from " << (readFromNFS ? "file (NFS)" : "database")
+       << endl;
+}
+
+/*****************************************************************************/ 
+void BeamSpotDipServer::bookHistograms(
+  DQMStore::IBooker&, const edm::Run&, const edm::EventSetup&)
+{
+  // do nothing
+}
+
+/*****************************************************************************/ 
+void BeamSpotDipServer::dqmBeginRun(
+  const edm::Run& r, const edm::EventSetup&)
+{
+  cout << "[Dip] begin run " << r.run() << endl;
```

Comment:
couts are not recommended/allowed, please use edm::Log instead

Reply:
ok